rasterio/plot.py

```diff
@@ -50,6 +50,9 @@ def show(source, cmap='gray', with_bounds=True):
         rather than pixel coordinates. Only works when source is
         (raster dataset, bidx).
     """
+    if plt is None:  # pragma: no cover
+        raise ImportError("Could not import matplotlib")
+
```

Comment:
Instead of catching the `ImportError` at the top of the module and then reraising it here if `plt` is `None`, how about turning https://github.com/mapbox/rasterio/issues/741 into a `get_plt()` function? Call `get_plt()` at line 53 above and there's no need to check the value of `plt`.